packages/world/src/modules/keysintable/KeysInTableModule.sol

```diff
@@ -0,0 +1,91 @@
+ * This mapping is stored in a table registered by the module at the `targetTableId` provided in the
+ * install methods arguments.
+ *
+ * Note: if a table with composite keys is used, only the first key is indexed
+ *
+ * Note: this module currently expects to be `delegatecalled` via World.installRootModule.
+ * Support for installing it via `World.installModule` depends on `World.callFrom` being implemented.
+ */
+contract KeysInTableModule is IModule, WorldContext {
+  using ResourceSelector for bytes32;
+
+  // The KeysInTableHook is deployed once and infers the target table id
+  // from the source table id (passed as argument to the hook methods)
+  KeysInTableHook immutable hook = new KeysInTableHook();
+
+  function getName() public pure returns (bytes16) {
+    return bytes16("index");
+  }
+
+  function grantAccess(IBaseWorld world, bytes32 targetTableSelector) internal {
+    // Grant the hook access to the target table
+    world.grantAccess(targetTableSelector.getNamespace(), targetTableSelector.getName(), address(hook));
+  }
+
+  function install(bytes memory args) public override {
+    // Extract source table id from args
+    bytes32 sourceTableId = abi.decode(args, (bytes32));
+
+    bytes32 targetTableSelector1 = getTargetTableSelector(MODULE_NAMESPACE, sourceTableId);
+    bytes32 targetTableSelector2 = getTargetTableSelector(KEYS_LENGTH_NAMESPACE, sourceTableId);
+    bytes32 targetTableSelector3 = getTargetTableSelector(USED_KEYS_NAMESPACE, sourceTableId);
+
+    IBaseWorld world = IBaseWorld(_world());
+
+    // Register the target table
+    world.registerTable(
+      targetTableSelector1.getNamespace(),
+      targetTableSelector1.getName(),
+      KeysInTable.getSchema(),
+      KeysInTable.getKeySchema()
+    );
+    world.registerTable(
+      targetTableSelector2.getNamespace(),
+      targetTableSelector2.getName(),
+      KeysInTableLength.getSchema(),
+      KeysInTableLength.getKeySchema()
+    );
+    world.registerTable(
+      targetTableSelector3.getNamespace(),
+      targetTableSelector3.getName(),
+      UsedKeysIndex.getSchema(),
+      UsedKeysIndex.getKeySchema()
+    );
+
+    // Register metadata for the target table
+    (string memory tableName1, string[] memory fieldNames1) = KeysInTable.getMetadata();
+    world.setMetadata(targetTableSelector1.getNamespace(), targetTableSelector1.getName(), tableName1, fieldNames1);
+    (string memory tableName2, string[] memory fieldNames2) = KeysInTableLength.getMetadata();
+    world.setMetadata(targetTableSelector2.getNamespace(), targetTableSelector2.getName(), tableName2, fieldNames2);
+    (string memory tableName3, string[] memory fieldNames3) = UsedKeysIndex.getMetadata();
+    world.setMetadata(targetTableSelector3.getNamespace(), targetTableSelector3.getName(), tableName3, fieldNames3);
+
+    grantAccess(world, targetTableSelector1);
+    grantAccess(world, targetTableSelector2);
+    grantAccess(world, targetTableSelector3);
```

Comment:
why is this needed?

Reply:
This grants the hook access to the target tables - without it I get an `AccessDenied` error.

Reply:
hm interesting, the world should have access to everything by default (see #609)